text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,799 @@
+  - Constructing a Resource for the SDK from detectors.
+  - Dealing with conflicts between detectors.
+  - Providing SDK-internal access to detected Resources for reporting via Log signal on configured LogProviders.
+  - *(new) Managing Entity changes during SDK lifetime, specifically dealing with entities that have lifetimes shorter than the SDK*
+
+#### Resource Coordinator
+
+The SDK Resource Coordinator is responsible for running all configured Resource and Entity Detectors.  There will be some (user-controlled, otel default) priority order to these.
+
+- The Resource Coordinator will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource (See [algorithm here](#entity-merging-and-resource)).
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it. SDKs will no longer fill out SchemaURL on Resource. Additionally, as no (non-service) Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB ["Env Variable Entity Detector"](#environment-variable-detector) will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
```

Comment:
This comment is caused by using an obscure language optimized for hard realtime programming. In Java, the result would simply be an ArrayList of entities and throw some DetectionException where applicable. The API proposal would be much clearer if any language with exceptions were chosen.

Reply:
We're not picking one language over another, and yes -> in java you can throw exceptions.

Reply:
That's not my point. It could also be written in some pseudo code with exceptions letting exception-averse languages pick their favorite error propagation mechanism. The point is, that the statement here is only required because one of these mechanisms is chosen over the more general concept.

Reply:
This is the second such comment about this code, the first asking for a go example (which is one I added), now you're asking for an exception throwing example.

I'm not picking exception vs. non-exception throwing languages here. The code is an example interface, you haven't shown me that this makes it difficult to understand or interpret, given your suggestion shows exactly how to translate into exception throwing languages.

I think the main point here is to *expliclty* show that the whole set of entities is returned or an error, which is important to make for non-exception languages where that isn't an implciit assumption.  I would do this even in pseudocode.

Reply:
Maybe we should change the example to pseudo code to avoid wasting time on programming language discussions.

Reply:
I'm closing this as resolved - won't fix.  If there's an aspect of the design that is struggling to be communicated, I'll fix it.  The comment implies that's not the case.